### PR TITLE
Add signatureToVeloxType API

### PR DESCRIPTION
**Pull request by duanmeng**
Add an API to transform the `TypeSignature` to Velox Type.

Details in https://github.com/facebookincubator/velox/discussions/7502

### CODE REVIEW

File: velox/expression/FunctionSignature.cpp [hunk 1]
```diff
@@ -139,6 +139,48 @@ TypeSignature parseTypeSignature(const std::string& signature) {
   return TypeSignature(baseName, std::move(nestedTypes));
 }
 
+TypePtr signatureToVeloxType(const TypeSignature& typeSignature) {
+  const auto& baseName = typeSignature.baseName();
+  if (baseName == "tinyint") {
```
Comment (mbasmanova) [inline velox/expression/FunctionSignature.cpp:144]: Can we use mapNameToTypeKind here?

File: velox/expression/FunctionSignature.cpp [hunk 2]
```diff
@@ -139,6 +139,48 @@ TypeSignature parseTypeSignature(const std::string& signature) {
...
+    return TINYINT();
+  } else if (baseName == "smallint") {
+    return SMALLINT();
+  } else if (baseName == "integer") {
```
Comment (mbasmanova) [inline velox/expression/FunctionSignature.cpp:148]: We may need to support Presto-specific types as well, i.e. JSON, TIMESTAMP WITH TIME ZONE.

Maybe we can use these APIs:

```
/// Returns true built-in or custom type with specified name exists.
bool hasType(const std::string& name);

/// Returns built-in or custom type with specified name and child types.
/// Returns nullptr if type with specified name doesn't exist.
TypePtr getType(
    const std::string& name,
    const std::vector<TypeParameter>& parameters);
```

File: velox/expression/FunctionSignature.cpp [hunk 3]
```diff
@@ -139,6 +139,48 @@ TypeSignature parseTypeSignature(const std::string& signature) {
...
+  } else if (baseName == "bigint") {
+    return BIGINT();
+  } else if (baseName == "float") {
+    return REAL();
+  } else if (baseName == "double") {
+    return DOUBLE();
+  } else if (baseName == "timestamp") {
+    return TIMESTAMP();
+  } else if (baseName == "decimal") {
+    const auto& parameters = typeSignature.parameters();
```
Comment (mbasmanova) [inline velox/expression/FunctionSignature.cpp:159]: See Type::create in Type.cpp. Maybe we could use something similar.

```
TypePtr Type::create(const folly::dynamic& obj) 
```

File: velox/expression/FunctionSignature.cpp (hunk 1 above)
Comment (duanmeng) [reply velox/expression/FunctionSignature.cpp:144]: It seems that the `kTypeStringMap` does not contain *decimal*.

**Comment by pedroerp** [comment]
@duanmeng I might be missing something here, but my mental model is that you cannot convert a signature into a concrete type without type biding. For example how would you convert a signature such as "array(T)" into a type? 

I think what you want to add is a "type parser", but then I think this file would not be the best place for it. Maybe that should be part of Type.[h|cpp] or even a new TypeParser.[h|cpp] unit?

File: velox/expression/FunctionSignature.cpp [hunk 4]
```diff
@@ -139,6 +139,48 @@ TypeSignature parseTypeSignature(const std::string& signature) {
   return TypeSignature(baseName, std::move(nestedTypes));
 }
 
+TypePtr signatureToVeloxType(const TypeSignature& typeSignature) {
```
Comment (kgpai) [inline velox/expression/FunctionSignature.cpp:142]: curious: Do we support variadic, generic types here ?

**Comment by duanmeng** [comment]
> I think what you want to add is a "type parser"

@pedroerp Yes, I want to add a *type parser* without type binding to parse the column type string of the response from the Presto, which is a blocker of porting PrestoQueryRunner to Velox. This PR could be closed as @majetideepak mentioned that he already had a PR to add this parser and it's more appropriate, see discussion #7502.
cc @mbasmanova @majetideepak @kgpai

File: velox/expression/FunctionSignature.cpp (hunk 4 above)
Comment (duanmeng) [reply velox/expression/FunctionSignature.cpp:142]: No, it has no binding.

**Pull request closed without merging**